components/org.wso2.carbon.identity.scim.common/src/main/java/org/wso2/carbon/identity/scim/common/utils/AdminAttributeUtil.java

```diff
@@ -64,20 +65,26 @@ public static void updateAdminUser(int tenantId, boolean validateSCIMID) throws
                             getTenantUserRealm(tenantId).getUserStoreManager();
             if (log.isDebugEnabled()) {
                 log.debug("SCIM enable in Userstore level : " + userStoreManager.isSCIMEnabled() + ", for "
-                          + "Tenant ID : " + tenantId + ", validating for the existing SCIM ID : " + validateSCIMID);
+                        + "Tenant ID : " + tenantId + ", validating for the existing SCIM ID : " + validateSCIMID);
             }
+
+            IdentityEventListenerConfig identityEventListenerConfig = IdentityUtil.readEventListenerProperty(
+                    "org.wso2.carbon.user.core.listener.UserOperationEventListener",
+                    "org.wso2.carbon.identity.scim.common.listener.SCIMUserOperationListener"
+            );
+
             //User store level property to enable/disable SCIM
-            if (userStoreManager.isSCIMEnabled()) {
+            if ("true".equals(identityEventListenerConfig.getEnable()) && userStoreManager.isSCIMEnabled()) {
```

Comment:
can identityEventListenerConfig become null? if so let;s add a null check

Reply:
Fixed in commit 9a0b2a8674ca20efdb3e4e5459f0b7e7e629dd8d